### PR TITLE
feat: Implements #345 a heatmap plot

### PR DESCRIPTION
**Pull request by mayurkmmt**

### CODE REVIEW

File: tests/plots/test_heatmap.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+    # Verify kwargs are passed to imshow
+    images = result_ax.get_images()
+    assert len(images) > 0
+    assert images[0].get_alpha() == test_alpha
+
+
+@pytest.mark.parametrize("shape", [(1, 5), (5, 1), (1, 1)])
+def test_plot_edge_case_dimensions(shape):
+    """Test heatmap with edge case DataFrame dimensions."""
```
Comment (murray-ds) [inline tests/plots/test_heatmap.py]: Can you make the docstring a little more descriptive please. I'm not sure what the "edge cases" are
